tsl/test/expected/continuous_aggs_union_view-11.out

```diff
@@ -0,0 +1,393 @@
+   GROUP BY _materialized_hypertable_2.time_bucket                                                                                                                                                +
+ UNION ALL                                                                                                                                                                                        +
+  SELECT time_bucket('@ 1 day'::interval, metrics."time") AS time_bucket,                                                                                                                         +
+     avg(metrics.value) AS avg                                                                                                                                                                    +
+    FROM metrics                                                                                                                                                                                  +
+   WHERE metrics."time" >= COALESCE(_timescaledb_internal.to_timestamp(_timescaledb_internal.cagg_watermark(1)), '-infinity'::timestamp with time zone)                                           +
+   GROUP BY (time_bucket('@ 1 day'::interval, metrics."time"));
+(1 row)
+
+SELECT time_bucket,avg FROM metrics_summary ORDER BY 1;
+         time_bucket          | avg 
+------------------------------+-----
+ Fri Dec 31 16:00:00 1999 PST | 0.5
+(1 row)
+
+-- downgrade view to non-union view
+ALTER VIEW metrics_summary SET (timescaledb.materialized_only=true);
+-- this should be view without union
+SELECT pg_get_viewdef('metrics_summary',true);
+                                                                                          pg_get_viewdef                                                                                           
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+  SELECT _materialized_hypertable_2.time_bucket,                                                                                                                                                  +
+     _timescaledb_internal.finalize_agg('avg(double precision)'::text, NULL::name, NULL::name, '{{pg_catalog,float8}}'::name[], _materialized_hypertable_2.agg_2_2, NULL::double precision) AS avg+
+    FROM _timescaledb_internal._materialized_hypertable_2                                                                                                                                         +
+   GROUP BY _materialized_hypertable_2.time_bucket;
+(1 row)
+
+SELECT time_bucket,avg FROM metrics_summary ORDER BY 1;
+ time_bucket | avg 
+-------------+-----
+(0 rows)
+
+-- upgrade view to union view again
+ALTER VIEW metrics_summary SET (timescaledb.materialized_only=false);
```

Comment:
Do we need to adda test to continuous_aggs_permissions.sql to alter view if not view owner. Is that needed or already covered by existing tests?

Reply:
alter view was covered but i added explicit test for materialized_only